chips/atmega328p-hal/src/lib.rs

```diff
@@ -93,9 +169,62 @@ pub mod spi {
     }
 }
 
+#[cfg(feature = "atmega328pb")]
+pub mod spi {
+    //! Implementation of the Rust Embedded-HAL SPI FullDuplex trait for AVR.
+    //!
+    //! The interface can be instantiated with the `new` method, and used directly
+    //! or passed into a driver.  Example usage:
+    //!
+    //! ```
+    //! pins.d10.into_output(&mut pins.ddr);// SS must be set to output mode
+    //! // create SPI interface
+    //! let mut spi = Spi0::new(
+    //!     dp.SPI,// SPI peripheral
+    //!     pins.d13.into_output(&mut pins.ddr),// SCLK
+    //!     pins.d11.into_output(&mut pins.ddr),// MOSI output pin
+    //!     pins.d12.into_pull_up_input(&mut pins.ddr),// MISO input pin
+    //!     Settings::default(),
+    //! );
+    //!
+    //! // Send a byte
+    //! let sent = 0b10101010;
+    //! spi.send(sent).unwrap();
+    //! let response = spi.read().unwrap();
+    //! ```
+    //! In the example above, all of the settings are left at the default.  You can
+    //! also instantiate a Settings object with the other options available.
+
+    use crate::port::{portb, portc, porte};
+    pub use avr_hal_generic::spi::*;
+
+    avr_hal_generic::impl_spi! {
+        pub struct Spi0 {
+            peripheral: crate::pac::SPI0,
+            pins: {
+                sclk: portb::PB5,
+                mosi: portb::PB3,
+                miso: portb::PB4,
+            }
+        }
+    }
+
+    avr_hal_generic::impl_spi! {
+        pub struct Spi1 {
+            peripheral: crate::pac::SPI1,
+            pins: {
+                sclk: portc::PC1,
+                mosi: porte::PE3,
+                miso: portc::PC0,
+            }
+        }
+    }
+}
+
 /// Serial interface using USART
 pub mod usart {
```

Comment:
Please feature-gate all modules that aren't already feature-gated by your changes behind `device-selected`.  Otherwise, when forgetting to select a device, you'd get a ton of compiler errors which just hide the one interesting one generated by the `compile_error!()` above.

```suggestion
#[cfg(feature = "device-selected")]
pub mod usart {
```

Reply:
Ah, I maybe didn't make it clear what I meant:  The other modules need the same
feature-guard:

```rust
#[cfg(feature = "device-selected")]
pub mod port;
#[cfg(feature = "device-selected")]
pub mod adc;
#[cfg(feature = "device-selected")]
pub mod pwm;
#[cfg(feature = "device-selected")]
pub mod wdt;

#[cfg(feature = "device-selected")]
pub mod prelude {
    pub use avr_hal_generic::prelude::*;
    pub use crate::port::PortExt as _;
}
```

Reply:
Your right. Added the gates and added missing pac documentation